dags/bgbb.py

```diff
@@ -0,0 +1,50 @@
+from datetime import datetime, timedelta
+
+from airflow import DAG
+from airflow.operators.moz_databricks import MozDatabricksSubmitRunOperator
+from utils.mozetl import mozetl_envvar
+
+default_args = {
+    "owner": "wbeard@mozilla.com",
+    "depends_on_past": False,
+    "start_date": datetime(2019, 2, 22),
+    "email": [
+        "telemetry-alerts@mozilla.com",
+        "wbeard@mozilla.com",
+        "amiyaguchi@mozilla.com",
+    ],
+    "email_on_failure": True,
+    "email_on_retry": True,
+    "retries": 2,
+    "retry_delay": timedelta(minutes=30),
+}
+
+# run on the 8th hour of the 1st day of the month
+dag = DAG("bgbb_fit", default_args=default_args, schedule_interval="0 8 1 * *")
+
+
+bgbb_fit = MozDatabricksSubmitRunOperator(
+    task_id="bgbb_fit",
+    job_name="Fit BGBB parameters",
+    execution_timeout=timedelta(hours=2),
+    instance_count=3,
+    env=mozetl_envvar(
+        "bgbb_fit",
+        {
+            "submission-date": "{{ ds }}",
```

Comment:
Reminder that `ds` will expand to the _beginning of the trigger period_ (in this case, I believe that means the 1st day of the previous month?) – will the job work correctly in light of that?

Reply:
This is somewhat confusing -- I'm guessing that the logs:

```2019-05-08 15:13:20          0 bgbb/params/submission_date_s3=2019-05-01/_SUCCESS```

are written as expected because I'm manually triggering them through the command-line? The predict job downstream will use the latest parameters that occur before the current date, so it would be out of sync by a month.

This job needs should be run like longitudinal -- is manipulating the macro the best way around this?

Reply:
Ugh, yeah, this is one of the unfortunate things about Airflow. Options: use the `next_ds` macro (but then the dates on manual triggers will be incorrect), or create a custom macro (the advantage there being that you can check whether the dag_run was manually triggered or not)

Reply:
I think this might change things for the manual triggers when using [next_ds in 1.10.1](https://github.com/apache/airflow/blob/master/UPDATING.md#semantics-of-next_dsprev_ds-changed-for-manually-triggered-runs). Apparently the next_ds macro will be the same as the execution date. There are a lot of things that are non-intuitive about this...